pkg/gardenlet/operation/botanist/kubeapiserverexposure.go

```diff
@@ -92,19 +93,24 @@ func (b *Botanist) setAPIServerServiceClusterIPs(clusterIPs []string) {
 		v1beta1constants.DeploymentNameKubeAPIServer,
 		b.Shoot.SeedNamespace,
 		func() *kubeapiserverexposure.SNIValues {
-			return &kubeapiserverexposure.SNIValues{
+			values := &kubeapiserverexposure.SNIValues{
 				Hosts: []string{
 					gardenerutils.GetAPIServerDomain(*b.Shoot.ExternalClusterDomain),
 					gardenerutils.GetAPIServerDomain(b.Shoot.InternalClusterDomain),
 				},
-				APIServerProxy: &kubeapiserverexposure.APIServerProxy{
-					APIServerClusterIP: b.APIServerClusterIP,
-				},
 				IstioIngressGateway: kubeapiserverexposure.IstioIngressGateway{
 					Namespace: b.IstioNamespace(),
 					Labels:    b.IstioLabels(),
 				},
 			}
+
+			if !features.DefaultFeatureGate.Enabled(features.RemoveAPIServerProxyLegacyPort) {
```

Comment:
Can you invert the logic to make it more readable? Or at least add a comment explaining that this cleans up the unused/unreachable `EnvoyFilter` acting on the removed API server proxy legacy port?